### PR TITLE
Bug fix of substitute holidays in Japan

### PR DESCRIPTION
**Pull request by shohirose**
Substitute holidays for Reiwa Emperor's Birthday (Feb 23) after 2020 are added. In addition, a test for substitute holidays in Japan between 2024 and 2040 is added.

### CODE REVIEW

File: test/countries/test_japan.py [hunk 1]
```diff
@@ -168,3 +168,34 @@ def test_invalid_years(self):
...
+        self.assertIn(date(2033, 3, 21), self.holidays)
+        self.assertIn(date(2034, 1, 2), self.holidays)
+        self.assertIn(date(2035, 2, 12), self.holidays)
+        self.assertIn(date(2035, 4, 30), self.holidays)
+        self.assertIn(date(2035, 9, 24), self.holidays)
+        self.assertIn(date(2036, 5, 6), self.holidays)
+        self.assertIn(date(2036, 11, 24), self.holidays)
+        self.assertIn(date(2037, 5, 6), self.holidays)
+        self.assertIn(date(2040, 1, 2), self.holidays)
+        self.assertIn(date(2040, 4, 30), self.holidays)
```
Comment (kasya) [inline test/countries/test_japan.py:201]:
```suggestion
        for dt in (
            (2024, 2, 12),
            (2024, 5, 6),
            (2024, 8, 12),
            (2024, 9, 23),
            (2024, 11, 4),
            (2025, 2, 24),
            (2025, 5, 6),
            (2025, 11, 24),
            (2026, 5, 6),
            (2027, 3, 22),
            (2029, 2, 12),
            (2029, 4, 30),
            (2029, 9, 24),
            (2030, 5, 6),
            (2030, 8, 12),
            (2030, 11, 4),
            (2031, 2, 24),
            (2031, 5, 6),
            (2031, 11, 24),
            (2033, 3, 21),
            (2034, 1, 2),
            (2035, 2, 12),
            (2035, 4, 30),
            (2035, 9, 24),
            (2036, 5, 6),
            (2036, 11, 24),
            (2037, 5, 6),
            (2040, 1, 2),
            (2040, 4, 30),
        ):
            self.assertIn(date(*dt), self.holidays)
```

**Review by kasya — APPROVED** [review]
LGTM

File: test/countries/test_japan.py (hunk 1 above)
Comment (shohirose) [reply test/countries/test_japan.py:201]: Thanks for refactoring! @kasya

Comment (kasya) [reply test/countries/test_japan.py:201]: > Thanks for refactoring! @kasya

No problem 👍🏻   If you'd like to apply these changes though, you need to click on "Commit suggestions" button under my suggested code as you're an author of this PR.

**Comment by dr-prodigy** [comment]
Hi @shohirose @kasya thank you both!
Would you please update this PR with tests coverage for the new substitute holidays?
Thx KUTGW!

**Comment by shohirose** [comment]
Hi @dr-prodigy . Thank you for your comments.
All of the substitute holidays in Japan between 1949 and 2050 are added to the test.
In addition, other holidays are tested more thoroughly.

**Pull request merged**